RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc

```diff
@@ -53,6 +55,22 @@ namespace {
     return ddd;
   }
 
+  inline const HGCalDDDConstants* get_ddd(const CaloSubdetectorGeometry* geom,
+					  const HGCSiliconDetId& detid) {
+    const HGCalGeometry* hg = dynamic_cast<const HGCalGeometry*>(geom);
```

Comment:
why cannot you just get a handle to the constants from an event setup?

Reply:
it looks like the response to this comment on the dynamic_cast is "it was there before" for the other methods.
However, I agree that this can be improved.
A somewhat daring way is to use a reinterpret_cast

Reply:
Please do not use a reinterpret_cast. A static_cast would be sufficient.